src/rdkafka.h

```diff
@@ -1377,13 +1378,19 @@ void rd_kafka_conf_set_consume_cb (rd_kafka_conf_t *conf,
  *
  * Without a rebalance callback this is done automatically by librdkafka
  * but registering a rebalance callback gives the application flexibility
- * in performing other operations along with the assinging/revocation,
+ * in performing other operations along with the assigning/revocation,
  * such as fetching offsets from an alternate location (on assign)
  * or manually committing offsets (on revoke).
  *
  * @remark The \p partitions list is destroyed by librdkafka on return
  *         return from the rebalance_cb and must not be freed or
  *         saved by the application.
+ *
+ * @remark The \p partitions list should not be modified for instance by
```

Comment:
It is actually okay to modify the partitions list to set initial offsets for each partition.
So maybe reword this as a precaution to only deliberately modifying the partitions list if the initial starting offsets should be changed. Otherwise pass a copy of the partitions list (using rd_kafka_topic_partition_list_copy() to functions which may modify it.

nit: position() is the current consume position, it is typically outdated in ASSIGN_PARTITIONS.